Source/SwiftLintFramework/Rules/LineLengthRule.swift

```diff
@@ -31,15 +31,37 @@ public struct LineLengthRule: ConfigurationProviderRule, SourceKitFreeRule {
     )
 
     public func validate(file: File) -> [StyleViolation] {
-        let minValue = configuration.params.map({ $0.value }).min(by: <)
+        let minValue = configuration.params.map({ $0.value }).min(by: <) ?? Int.max
+        let swiftDeclarationKindsByLine: [[SwiftDeclarationKind]] = file.swiftDeclarationKindsByLine() ?? []
+        let syntaxKindsByLine: [[SyntaxKind]] = file.syntaxKindsByLine() ?? []
         return file.lines.flatMap { line in
             // `line.content.characters.count` <= `line.range.length` is true.
             // So, `check line.range.length` is larger than minimum parameter value.
             // for avoiding using heavy `line.content.characters.count`.
-            if line.range.length < minValue! {
+            if line.range.length < minValue {
                 return nil
             }
 
+            if configuration.ignoresFunctionDeclarations &&
+                line.index < swiftDeclarationKindsByLine.count {
+                let functionKinds = swiftDeclarationKindsByLine[line.index].filter { kind in
+                    SwiftDeclarationKind.functionKinds().contains(kind)
+                }
+                if !functionKinds.isEmpty {
+                    return nil
+                }
+            }
+
+            if configuration.ignoresComments &&
+                line.index < syntaxKindsByLine.count {
+                let lineCommentKinds = syntaxKindsByLine[line.index].filter {
+                    return SyntaxKind.commentKinds().contains($0)
+                }
+                if !lineCommentKinds.isEmpty {
```

Comment:
Actually, I don't know if this is the expected behavior. I'd expect that the line would be ignored if it only had comments.

A long line with a small comment in the end should trigger IMO.